src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,109 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static int capacity = 10;
+    private static final int ONE = 1;
+    private static final int ZERO = 0;
+    private T[] items;
+    private int size;
+
+    public ArrayList() {
+        items = (T[]) new Object[capacity];
+    }
+
     @Override
     public void add(T value) {
-
+        if (size == capacity) {
+            grow();
+        }
+        items[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index < ZERO || index > size) {
+            throw new ArrayListIndexOutOfBoundsException("This position is not exist in list");
+        }
+        if (size == capacity) {
+            grow();
+        }
+        T[] newItems = (T[]) new Object[capacity];
+        System.arraycopy(items, ZERO, newItems, ZERO, index);
+        newItems[index] = value;
+        System.arraycopy(items, index, newItems, index + ONE, size - index);
+        items = newItems;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        while (capacity < size + list.size()) {
+            grow();
+        }
+        T[] newItems = (T[]) new Object[capacity];
+        System.arraycopy(items, ZERO, newItems, ZERO, size);
+        System.arraycopy(list.toArray(), ZERO, newItems, size, list.size());
```

Comment:
```suggestion
        System.arraycopy(list.toArray(), ZERO, items, size, list.size());
```
you can operate with initial array "items"